experimental/packages/otlp-transformer/src/common/internal.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { SpanAttributes } from '@opentelemetry/api';
+import { IAnyValue, IKeyValue } from './types';
+
+export function toAttributes(
+  attributes: SpanAttributes
+): IKeyValue[] {
+  return Object.keys(attributes).map(key => toKeyValue(key, attributes[key]));
+}
+
+export function toKeyValue(
+  key: string,
+  value: unknown
+): IKeyValue {
+  return {
+    key: key,
+    value: toAnyValue(value),
+  };
+}
+
+export function toAnyValue(value: unknown): IAnyValue {
+  const t = typeof value;
+  if (t === 'string') return { stringValue: value as string };
+  if (t === 'number' && !Number.isInteger(value)) return { doubleValue: value as number };
+  if (t === 'number' && Number.isInteger(value)) return { intValue: value as number };
```

Comment:
optional minor performance improvement:
```js
  if (t === 'number') {
    if(Number.isInteger(value)) return { intValue: value as number }
    else { doubleValue: value as number }
  } 
```

This saves the redundant comparison to `'number'` and invoking `Number.isInteger` again when can be avioded.